cmake/CeleritasUtils.cmake

```diff
@@ -164,16 +164,41 @@ endfunction()
 function(celeritas_add_library target)
 
   set(_midsuf "")
+  set(_staticsuf "_static")
   celeritas_sources_contains_cuda(_contains_cuda ${ARGN})
 
-  if(NOT BUILD_SHARED_LIBS OR NOT CELERITAS_USE_CUDA OR NOT _contains_cuda)
+  # Whether we need the special code or not is actually dependent on information
+  # we don't have ... yet
+  # - whether the user request CUDA_SEPARABLE_COMPILATION
+  # - whether the library depends on a library with CUDA_SEPARABLE_COMPILATION code.
+  # I.e. this should really be done at generation time.
+  # So in the meantime we use CELERITAS_USE_VecGeom as a proxy.
+
+  if(NOT CELERITAS_USE_VecGeom OR NOT CELERITAS_USE_CUDA OR NOT _contains_cuda)
     add_library(${target} ${ARGN})
     return()
   endif()
 
+  cmake_parse_arguments(_ADDLIB_PARSE
+    "STATIC;SHARED;MODULE"
+    ""
+    ""
+    ${ARGN}
+  )
+  set(_lib_requested_type "SHARED")
+  set(_cudaruntime_requested_type "Shared")
+  if((NOT BUILD_SHARED_LIBS AND NOT _ADDLIB_PARSE_SHARED) OR _ADDLIB_PARSE_STATIC)
+    set(_lib_requested_type "STATIC")
+    set(_cudaruntime_requested_type "Static")
+    set(_staticsuf "")
+  endif()
+  # Do we need to do anything if the type is MODULE?
```

Comment:
Nope, we would only need MODULE if we're building a standalone shared library for loading with `dlopen`... god forbid. I don't think that would play well with our cuda  RDC infrastructure anyway.